packages/swap/src/routes/__tests__/v2/swap.test.ts

```diff
@@ -1168,6 +1199,227 @@ describe('server', () => {
       expect(req1.body).toStrictEqual(req2.body);
     });
 
+    it(`retrieves a swap from an onChain origin in ${StateV2.Swapping}`, async () => {
+      const accountId = 'cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh';
+      const requestedEvent = createOnChainSwapRequestedEvent(accountId);
+
+      await processEvents([requestedEvent, swapEventMap['Swapping.SwapScheduled']]);
+
+      const { body, status } = await request(server).get(
+        `/v2/swaps/${requestedEvent.args.swapRequestId}`,
+      );
+      expect(status).toBe(200);
+      expect(body).toMatchSnapshot();
+    });
+
+    it(`retrieves a swap from an onChain origin in ${StateV2.Completed}`, async () => {
+      const accountId = 'cFNzKSS48cZ1xQmdub2ykc2LUc5UZS2YjLaZBUvmxoXHjMMVh';
+      const requestedEvent = createOnChainSwapRequestedEvent(accountId);
+
+      const executedEvent = clone(swapEventMap['Swapping.SwapExecuted']);
+      executedEvent.args = {
+        ...executedEvent.args,
+        brokerFee: '0',
+      };
+
+      await processEvents([
+        requestedEvent,
+        swapEventMap['Swapping.SwapScheduled'],
+        executedEvent,
+        swapEventMap['Swapping.SwapRequestCompleted'],
+      ]);
+
+      const { body, status } = await request(server).get(
+        `/v2/swaps/${requestedEvent.args.swapRequestId}`,
+      );
+
+      expect(status).toBe(200);
+      expect(body).toMatchSnapshot();
+    });
+
+    it(`retrieves a DCA swap from an onChain origin in ${StateV2.Completed}`, async () => {
+      const accountId = '0x640dfbca7473dd212d3c9b9815cd32dcc83b2a9a099c91369110609199b0f374';
+
+      // requested
+      const requestedEvent = {
+        name: 'Swapping.SwapRequested',
+        id: '0000004123-000984-b7c4d',
+        blockId: '0000015161-d719c',
+        indexInBlock: 21,
+        args: {
+          origin: {
+            value: accountId,
+            __kind: 'OnChainAccount',
+          },
+          brokerFees: [],
+          inputAsset: {
+            __kind: 'Usdc',
+          },
+          inputAmount: '5000000000',
+          outputAsset: {
+            __kind: 'Usdt',
+          },
+          requestType: {
+            __kind: 'Regular',
+            outputAction: {
+              __kind: 'CreditOnChain',
+              accountId,
+            },
+          },
+          dcaParameters: {
+            chunkInterval: 2,
+            numberOfChunks: 2,
+          },
+          swapRequestId: '4',
+          refundParameters: {
+            minPrice: '0',
+            retryDuration: 100,
+            refundDestination: {
+              value: accountId,
+              __kind: 'InternalAccount',
+            },
+          },
+        },
+      };
+
+      // scheduled 1
+      const legOneScheduled = {
```

Comment:
These are chunks instead of legs
```suggestion
      const chunkOneScheduled = {
```

Reply:
Renamed